dev/com.ibm.ws.session.cache/src/com/ibm/ws/session/store/cache/CacheHashMap.java

```diff
@@ -112,60 +128,84 @@
+
+        sessionMetaCache = cacheStoreService.cacheManager.getCache(metaCacheName, String.class, ArrayList.class);
+        boolean create;
+        if (create = sessionMetaCache == null) {
+            if (trace && tc.isDebugEnabled())
+                tcReturn(cacheStoreService.tcCacheManager, "getCache", "null");
 
-        sessionInfoCache = cacheStoreService.cacheManager.getCache(infoCacheName, String.class, ArrayList.class);
-        if (sessionInfoCache == null) {
             @SuppressWarnings("rawtypes")
             MutableConfiguration<String, ArrayList> config = new MutableConfiguration<String, ArrayList>()
                             .setTypes(String.class, ArrayList.class)
                             .setExpiryPolicyFactory(EternalExpiryPolicy.factoryOf());
-            if (cacheStoreService.cachingProvider.isSupported(OptionalFeature.STORE_BY_REFERENCE))
+            if (cacheStoreService.supportsStoreByReference)
                 config = config.setStoreByValue(false);
             try {
-                sessionInfoCache = cacheStoreService.cacheManager.createCache(infoCacheName, config);
-
                 if (trace && tc.isDebugEnabled())
-                    Tr.debug(tc, "Created a new session info cache", sessionInfoCache);
+                    tcInvoke(cacheStoreService.tcCacheManager, "createCache", metaCacheName, config);
+
+                sessionMetaCache = cacheStoreService.cacheManager.createCache(metaCacheName, config);
             } catch (CacheException x) {
-                sessionInfoCache = cacheStoreService.cacheManager.getCache(infoCacheName, String.class, ArrayList.class);
-                if (sessionInfoCache == null)
+                create = false;
+                if (trace && tc.isDebugEnabled()) {
+                    tcReturn(cacheStoreService.tcCacheManager, "createCache", x);
+                    tcInvoke(cacheStoreService.tcCacheManager, "getCache", metaCacheName, "String", "ArrayList");
+                }
+                sessionMetaCache = cacheStoreService.cacheManager.getCache(metaCacheName, String.class, ArrayList.class);
+                if (sessionMetaCache == null)
                     throw x;
             }
         }
 
-        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
-            Tr.debug(tc, "Using session info cache", sessionInfoCache);
+        tcSessionMetaCache = "MetaCache" + Integer.toHexString(System.identityHashCode(sessionMetaCache));
+        if (trace && tc.isDebugEnabled())
+            tcReturn(cacheStoreService.tcCacheManager, create ? "createCache" : "getCache", tcSessionMetaCache, sessionMetaCache);
+
+        // Session Attributes Cache
 
-        String propCacheName = new StringBuilder(24 + a.length()).append("com.ibm.ws.session.prop.").append(a).toString();
+        String propCacheName = new StringBuilder(24 + a.length()).append("com.ibm.ws.session.attr.").append(a).toString();
```

Comment:
Update the variable name here to attrCacheName to match the updated cache name.

Reply:
updated here and also updated other variables using the term property instead of attribute